src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy64NativeMethods.cs

```diff
@@ -0,0 +1,96 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using MongoDB.Driver.Core.NativeLibraryLoader;
+
+namespace MongoDB.Driver.Core.Compression.Snappy
+{
+    internal static class Snappy64NativeMethods
+    {
+        private static readonly Lazy<LibraryLoader> __libraryLoader;
+        private static readonly Lazy<Delegates64.snappy_compress> __snappy_compress;
+        private static readonly Lazy<Delegates64.snappy_max_compressed_length> __snappy_max_compressed_length;
+        private static readonly Lazy<Delegates64.snappy_uncompress> __snappy_uncompress;
+        private static readonly Lazy<Delegates64.snappy_uncompressed_length> __snappy_uncompressed_length;
+        private static readonly Lazy<Delegates64.snappy_validate_compressed_buffer> __snappy_validate_compressed_buffer;
+
+        static Snappy64NativeMethods()
+        {
+            __libraryLoader = new Lazy<LibraryLoader>(() => new LibraryLoader(GetLibraryLocator), isThreadSafe: true);
+
+            __snappy_compress = CreateLazyForDelegate<Delegates64.snappy_compress>(nameof(snappy_compress));
+            __snappy_max_compressed_length = CreateLazyForDelegate<Delegates64.snappy_max_compressed_length>(nameof(snappy_max_compressed_length));
+            __snappy_uncompress = CreateLazyForDelegate<Delegates64.snappy_uncompress>(nameof(snappy_uncompress));
+            __snappy_uncompressed_length = CreateLazyForDelegate<Delegates64.snappy_uncompressed_length>(nameof(snappy_uncompressed_length));
+            __snappy_validate_compressed_buffer = CreateLazyForDelegate<Delegates64.snappy_validate_compressed_buffer>(nameof(snappy_validate_compressed_buffer));
+        }
+
+        // public static methods
```

Comment:
If you have `// public static methods` you should have similar comments for the static fields and the static constructor.